communicator/ssh/key_pair.go

```diff
@@ -60,10 +60,10 @@ type KeyPair struct {
 // KeyPairFromPrivateKey returns a KeyPair loaded from an existing private key.
 //
 // Supported key pair types include:
-// 	- DSA
-// 	- ECDSA
-// 	- ED25519
-// 	- RSA
+//   - DSA
```

Comment:
I understand that DSA is marked as `nolint` here to remove the deprecation warning from the linters, which is acceptable in my opinion, however I would maybe argue in favour of warning users not to use DSA anymore since it's widely recognized as non-secure, and future versions of Go may remove the `crypto/dsa` package altogether.

Maybe consider adding the following message to this comment?

```suggestion
//   - DSA (NOTE: deprecated from Go as DSA is not regarded as secure anymore, please consider RSA or ED25519 instead)
```

Reply:
Good idea, makes sense to me.